modules/git/repo_base_nogogit.go

```diff
@@ -27,10 +27,12 @@ type Repository struct {
 
 	gpgSettings *GPGSettings
 
+	batchInUse  bool
 	batchCancel context.CancelFunc
 	batchReader *bufio.Reader
 	batchWriter WriteCloserError
 
+	checkInUse  bool
```

Comment:
Maybe these "bool" flags should be reset to false in `Close`

Reply:
Yup, it shouldn't be reused. But the old code tried to make it reusable (reset everything), so I think it's good to keep consistent with old code at the moment.